cruise.umple/src/UmpleInternalParser_CodeClass.ump

```diff
@@ -2847,10 +2814,58 @@ class UmpleInternalParser
       }
     }
 
-    for(String operation : operationName.split(","))
+    return methodNames;
+  }
+
+  
+  private void checkCodeInjectionValidityCustom(Token injectToken, UmpleClass uClass) {
+    List<Method> methods = uClass.getMethods();
+    String[] allOperations = getOperationName(injectToken).split(",");
+    List<String> allParameters = getOperationsParameters(injectToken);
+
+    String operationSource = getOperationSource(injectToken);
+    if("".equals(operationSource)) {
+      operationSource = "generated";
+    }
+
+    for(int opInd = 0; opInd < allOperations.length; opInd++)
     {
+      String operation = allOperations[opInd];
+      String currentParameters = allParameters.get(opInd);
+
       Boolean matches = false;
 
+      for(Method method : methods) 
+      { 
+        boolean tmpMatches = uClass.matchOperationMethod(operation, method.getName()) && checkParameterMatch(method.getMethodParameters(), currentParameters);
+        matches = matches || tmpMatches;
+      }
+
+      if(!matches) {
+        getParseResult().addErrorMessage(new ErrorMessage(1012, injectToken.getPosition(), operation));
+      }
+    }
+  } 
+
+  private void checkCodeInjectionValidityAll(Token injectToken, UmpleClass uClass) {
+    List<Method> methods = uClass.getMethods();
+    ArrayList<String> methodNames = getAllMethodNames(uClass);
+    String[] allOperations = getOperationName(injectToken).split(",");
+    List<String> allParameters = getOperationsParameters(injectToken);
+
+    String operationSource = getOperationSource(injectToken);
+    if("".equals(operationSource)) {
+      operationSource = "generated";
+    }
+
+    for(int opInd = 0; opInd < allOperations.length; opInd++)
+    {
+      String operation = allOperations[opInd];
+      String currentParameters = allParameters.get(opInd);
+
+      Boolean matches = false;
```

Comment:
This should be `bool` not `Boolean`. The `Object` forces a box and unboxing everytime you use it as a primitive.